### PR TITLE
Page views per article

### PR DESCRIPTION
**Pull request by hborrelli1**
Display number of page views for each article on the authors bio page.

**How to Test:**
Navigate to beta.section.io/engineering-education/ (or pull down locally), and visit a few author bio pages. You should see the new page views icon with the number of page views at the bottom of each article listed in the authors bio page (screenshot below).

image

If interested the API repo is [here](https://github.com/section-io/ga-page-views-app).

### CODE REVIEW

File: layouts/authors/single.html [hunk 1]
```diff
@@ -198,16 +216,45 @@ <h3 class="title-3 xs-mb-20">{{ .Site.Data.globals.enged_cta.title }}</h3>
...
+      insightItems[i].addEventListener("touch", () => toggleToolTip(insightItems[i]))
+    }
+  document.addEventListener("DOMContentLoaded", async function() {
+    const getPageView = (url, currentUrl) => {
+      const urlRegex = /(http:\/\/localhost:1313\/)|(https:\/\/beta.section.io\/engineering-education\/)|(https:\/\/section.io\/engineering-education\/)/gi;
+      const trimmedUrl = url.replace(urlRegex, '');
+
+      return fetch(`https:\/\/api.www.section.io/pageviews/${trimmedUrl}`)
+        .then(res => res.json())
+        .then(data => {
```
Comment (mwickman) [inline layouts/authors/single.html:237]: Have you considered what happens when this fetch errors? UI Still OK?

File: layouts/authors/single.html [hunk 2]
```diff
@@ -198,16 +216,45 @@ <h3 class="title-3 xs-mb-20">{{ .Site.Data.globals.enged_cta.title }}</h3>
...
+      while (currentUrl <= urls.length - 1) {
+        await getPageView(urls[currentUrl], currentUrl);
+        currentUrl++;
+      }
+    }
 
-  for (let i = 0; i < insightItems.length; i++) {
-    insightItems[i].addEventListener("touch", () => toggleToolTip(insightItems[i]))
-  }
+    let urlsOfArticles = "{{ $articleUrlsList }}";
```
Comment (mwickman) [inline layouts/authors/single.html:254]: Prefer const over let if possible

File: layouts/authors/single.html [hunk 3]
```diff
@@ -198,16 +216,45 @@ <h3 class="title-3 xs-mb-20">{{ .Site.Data.globals.enged_cta.title }}</h3>
...
+            document.querySelector(`#page-views-count-${currentUrl}`).innerText = numPageViews;
+          }
+        });
+    }
+
+    const applyPageViews = async (urls) => {
+      let currentUrl = 0;
+
+      while (currentUrl <= urls.length - 1) {
+        await getPageView(urls[currentUrl], currentUrl);
```
Comment (mwickman) [inline layouts/authors/single.html:249]: Just remember you need a try/catch to handle await errors. You may not need to handle the error in your case.

**Review by mwickman — APPROVED** [review]

File: assets/scss/components/_blog-listing.scss [hunk 4]
```diff
@@ -188,12 +187,24 @@
           padding: 10px 0;
 
           span {
-            margin-top: 0;
+            margin: 0;
```
Comment (manibatra) [inline assets/scss/components/_blog-listing.scss:190]: You could use the shorthand here `margin: 0 0 0 10px`

File: layouts/authors/single.html (hunk 1 above)
Comment (hborrelli1) [reply layouts/authors/single.html:246]: By default the page views icon are now hidden. Only with a successful response will we display the page views. I'd rather not show them at all if they are just showing 0 (failed response).

File: layouts/authors/single.html (hunk 3 above)
Comment (hborrelli1) [reply layouts/authors/single.html:262]: Errors are being handled at the previous level.

**Pull request merged**